mypy/checker.py

```diff
@@ -1121,7 +1122,7 @@ def check_func_item(
                 raise RuntimeError("Not supported")
 
         self.dynamic_funcs.pop()
-        self.current_node_deferred = False
+        self.current_node_deferred = enclosing_node_deferred
```

Comment:
```suggestion
        self.current_node_deferred = not self.dynamic_funcs and enclosing_node_deferred
```

Shouldn't this flag be reset to False when leaving the topmost function? That should explain new primer hits